### DISCUSSION TITLE
Many vs single ImageUpdateAutomation

### DISCUSSION THREAD

**Discussion by yoan-myparcel** [fluxcd/flux2#2702] [Q&A] [answered]
# Question
Hey all, my question is is it recommended to use many ImageUpdateAutomation objects (one per microservice in a stack of 30 microservices), or one that updates across yamls of multiple microservices? 

Furthermore, is there any limit of how many ImageUpdateAutomation objects you can run? Are there any bad effects (on the performance of the flux controllers for example) if there are too many such objects?

Thanks for the great project, flux is awesome!

**Comment by stefanprodan** [comment, marked as answer]
ImageUpdateAutomations are meant to be used once per namespace, the controller applies all `ImagePolicies` in the same namespace as the `ImageUpdateAutomation` object.

**Reply by wvh** [reply]
Should we care about the performance of 50 or even 100 separate ImageUpdateAutomations?

I'm in the same boat, trying to run many copies of one application, and it makes more sense to keep all resources in their own namespace, but it feels weird to have more than one ImageUpdateAutomation. Clearly our mental model doesn't entirely fit the reality of how the controllers are implemented.

**Reply by kingdonb** [reply]
I think the guidance is meant to say, at most once per namespace. It is perfectly normal to have a single ImageUpdateAutomation per-cluster, or even one ImageUpdateAutomation per repository (with usages spread across many clusters/multiple environments)

You could have more than one ImageUpdateAutomation in a given namespace, if you need to apply changes in **multiple repositories** from the same ImagePolicies, without duplicating the policies or their associated ImageRepositories.

The main idea is that IAC requires write credentials for your git repository, and that's what we call a "high-value target" so you definitely wouldn't want to have a copy of them in every namespace, on every cluster. That would make it very difficult to keep them protected and isolated from threat-actors.
